include/xtensor/xtiny.hpp

```diff
@@ -0,0 +1,1925 @@
+    inline void
+    xtiny_impl<V, runtime_size, R>::assign(IT begin, IT end)
+    {
+        std::ignore = end;
+        XTENSOR_ASSERT_MSG(std::distance(begin, end) == static_cast<std::ptrdiff_t>(size()),
+            "xtiny_impl::assign(begin, end): size mismatch.");
+        for(size_type k=0; k<size(); ++k, ++begin)
+        {
+            (*this)[k] = static_cast<value_type>(*begin);
+        }
+    }
+
+    template <class V, class R>
+    inline auto
+    xtiny_impl<V, runtime_size, R>::operator[](size_type i) -> reference
+    {
+        return m_data[i];
+    }
+
+    template <class V, class R>
+    constexpr inline auto
+    xtiny_impl<V, runtime_size, R>::operator[](size_type i) const -> const_reference
+    {
+        return m_data[i];
+    }
+
+    template <class V, class R>
+    inline auto
+    xtiny_impl<V, runtime_size, R>::data() -> pointer
+    {
+        return &m_data[0];
+    }
+
+    template <class V, class R>
+    constexpr inline auto
+    xtiny_impl<V, runtime_size, R>::data() const -> const_pointer
+    {
+        return &m_data[0];
+    }
+
+    template <class V, class R>
+    constexpr inline auto
+    xtiny_impl<V, runtime_size, R>::size() const -> size_type
+    {
+        return m_size;
+    }
+
+    template <class V, class R>
+    constexpr inline auto
+    xtiny_impl<V, runtime_size, R>::max_size() const -> size_type
+    {
+        return m_size;
+    }
+
+    template <class V, class R>
+    constexpr inline auto
+    xtiny_impl<V, runtime_size, R>::capacity() const -> size_type
+    {
+        return m_size;
+    }
+
+    template <class V, class R>
+    inline auto
+    xtiny_impl<V, runtime_size, R>::begin() -> iterator
+    {
+        return m_data;
+    }
+
+    template <class V, class R>
+    constexpr inline auto
+    xtiny_impl<V, runtime_size, R>::begin() const -> const_iterator
+    {
+        return m_data;
+    }
+
+    template <class V, class R>
+    inline auto
+    xtiny_impl<V, runtime_size, R>::rbegin() -> reverse_iterator
+    {
+        return reverse_iterator(m_data+m_size);
+    }
+
+    template <class V, class R>
+    constexpr inline auto
+    xtiny_impl<V, runtime_size, R>::rbegin() const -> const_reverse_iterator
+    {
+        return const_reverse_iterator(m_data+m_size);
+    }
+
+    template <class V, class R>
+    inline void
+    xtiny_impl<V, runtime_size, R>::swap(xtiny_impl & other)
+    {
+        using std::swap;
+        swap(m_size, other.m_size);
+        swap(m_data, other.m_data);
+    }
+
+
+    /**************************************************/
+    /* xtiny_impl xbuffer_adaptor view implementation */
+    /**************************************************/
+
+    template <class V, class CP, class O, class A>
+    inline void
+    xtiny_impl<V, runtime_size, xbuffer_adaptor<CP, O, A>>::assign(size_type n, const value_type& v)
+    {
+        std::ignore = n;
+        XTENSOR_ASSERT_MSG(n == size(), "xtiny_impl::assign(n, v): size mismatch.");
```

Comment:
I think assign is supposed to take care of resizing the container (this is the behavior from std::vector).
The following works:

```cpp
std::vector<double> test;
test.assign({1,2,3,4,5});
assert(test.size() == 5);
```

Reply:
Yes, but recall that `xtiny` exists in order to *unify* the APIs of static and dynamic arrays. Since a static array's `assign` cannot resize, I had to stick to this behavior for dynamic arrays as well.

Reply:
For the same reason, `insert()` and `erase()` don't work in-place, but return a new `xtiny`.